### ISSUE TITLE
Diagnostics strings - translate or not?

### ISSUE THREAD

**Issue by petecooper** [textpattern/textpattern#1118]
See Uli's forum post for background: https://forum.textpattern.io/viewtopic.php?pid=308766#p308766 and @philwareham's follow-up request: https://forum.textpattern.io/viewtopic.php?pid=308778#p308778

I'm biased as an English speaker, but I am +1 for leaving them purely in English.

**Comment by bloatware** [comment]
These are not translated anymore and go into `mode.ini` file.

**Comment by uli-on** [comment]
These are certainly those strings appearing in the preflight check section on the Diagnostics page. Those should be duplicated when diagnostics strings remain in English, they're directed at the _user_.

**Comment by philwareham** [comment]
Only some of the diagnostic strings are used in the preflight report - those would have to be ascertained and the moved to the mode.ini file under a different group name I guess. @bloke whats your take in this?

**Comment by philwareham** [comment]
I'm keen to get this done before 4.7 beta if it needs doing. If *some* of the diagnostics strings can be left in English (basically, the strings that appear in the preflight check textarea) then that reduces the amount of work translators need to do, and makes the preflight report understandable by the wider community.

I would need to know:

1. A list of gTxt stings that are only used in preflight report (probably quite easy to ascertain from `diag.php`)
2. What the group name would be for these new strings when put into the `mode.ini` file (i.e. can it still be `[diag]` group there or another name entirely?

Then, obviously, one of you PHP guys would need to amend the code so that these strings are read from the correct location.

**Comment by Bloke** [comment]
This is a good idea. A list of gTxt strings is easy to scrape from pre-flight, as you say. As for the group name, sticking it under `[diag]` should work if we go with code like this:

``` php
function load_lang($lang, $events = null)
{
    global $production_status, $event;

    $textarray = Txp::get('\Textpattern\L10n\Lang')->load($lang, $events);

    if ($lang == LANG
        && ($production_status !== 'live' || $event === 'diag')
        && @$debug = parse_ini_file(txpath.DS.'mode.ini')
    ) {
        $textarray += (array)$debug;
        Txp::get('\Textpattern\L10n\Lang')->setPack($textarray);
    }

    return $textarray;
}
```

Untested.

**Comment by philwareham** [comment]
@bloke OK, I've opened a separate branch for this work. I can also take the opportunity to reword some of the diagnostic string names so there is some consistency. I'll let you know when the strings are in `mode.ini` and maybe you can do the PHP side of it?

**Comment by Bloke** [comment]
No worries. Should be a simple case of swapping out the `load_lang()` function in `txplib_misc.php` with the version I posted above. Feel free to test if you like.

**Comment by philwareham** [comment]
@bloke I've done all the Textpack work for this now. Just need you to put your code example into wherever and give it a quick test, then we can merge.

That'll be 26 strings that translators don't need to worry about again! :)

**Comment by philwareham** [comment]
Seems to already read these new texts from `mode.ini` in my tests without adding additional code?!

**Comment by Bloke** [comment]
They're probably still in your lang table. Need to remove english and try again.

**Comment by philwareham** [comment]
I'd already wiped the English and done that. Definitely reading the strings from `mode.ini` without additional code,

**Comment by Bloke** [comment]
Weird, it's not for me. It works with the new code in English, but when I switch to a language other than that, I get untranslated strings for some reason, even though it loads the `mode.ini`. I think it's to do with the tests or the fact it's not injectuing them in the right area. Will test more thoroughly later.

**Comment by philwareham** [comment]
Works fine for me- wiped English from the database, installed Spanish only and I get this (which is correct as far as I can see)...

![grab](https://user-images.githubusercontent.com/413665/35229652-db15168e-ff8b-11e7-9f91-01084e115cdf.png)

**Comment by Bloke** [comment]
And it works even if you put your site in Live mode? Wonder why I'm getting odd results...

**Comment by philwareham** [comment]
Oh, you're right - once the site is in Live mode the strings are missing. Sorry.

**Comment by Bloke** [comment]
Okay, definitely more testing required. I'll get it working.

**Comment by Bloke** [comment]
Should be fixed in 870868d.
